auraed/src/cri/runtime_service.rs

```diff
@@ -115,72 +128,115 @@ impl runtime_service_server::RuntimeService for RuntimeService {
-        let syscall = create_syscall();
-        let sandbox_builder = ContainerBuilder::new(
-            AURAE_SELF_IDENTIFIER.to_string(),
-            syscall.as_ref(),
-        );
-
-        // Spawn auraed here
-        // TODO Check if sandbox already exists?
-        let _spawned = spawn_auraed_oci_to(
-            Path::new(AURAE_BUNDLE_PATH)
-                .join(AURAE_SELF_IDENTIFIER)
-                .to_str()
-                .expect("recursive path"),
-            oci_builder.build().expect("building pod oci spec"),
-        );
-
-        let sandbox_id = metadata.name;
-        let mut sandbox = sandbox_builder
-            .with_root_path(Path::new(AURAE_PODS_PATH).join(sandbox_id.clone()))
-            .expect("Setting pods directory")
-            .as_init(Path::new(AURAE_BUNDLE_PATH).join(AURAE_SELF_IDENTIFIER))
-            .with_systemd(false)
-            .build()
-            .expect("failed building pod sandbox: ensure valid OCI spec and proper container starting point");
-
-        sandbox.start().expect("starting pod sandbox");
-
-        // TODO: Cache sandbox
+        let sandbox = {
+            // Initialize a new container builder with the AURAE_SELF_IDENTIFIER name as the "init" container running a recursive Auraed
+            let syscall = create_syscall();
+            let sandbox_builder = ContainerBuilder::new(
+                AURAE_SELF_IDENTIFIER.to_string(),
+                syscall.as_ref(),
+            );
+
+            // Spawn auraed here
+            // TODO Check if sandbox already exists and if it does, clean up to this point?
+            let _spawned = spawn_auraed_oci_to(
+                Path::new(AURAE_BUNDLE_PATH)
+                    .join(AURAE_SELF_IDENTIFIER)
+                    .to_str()
+                    .expect("recursive path"),
+                oci_builder.build().expect("building pod oci spec"),
+            );
+
+            let mut sandbox = Some(sandbox_builder
+                .with_root_path(Path::new(AURAE_PODS_PATH).join(sandbox_id.clone()))
+                .expect("Setting pods directory")
+                .as_init(Path::new(AURAE_BUNDLE_PATH).join(AURAE_SELF_IDENTIFIER))
+                .with_systemd(false)
+                .build()
+                .expect("failed building pod sandbox: ensure valid OCI spec and proper container starting point"));
+
+            sandbox
+                .as_mut()
+                .expect("a sandbox")
+                .start()
+                .expect("starting pod sandbox");
+            sandbox
+        };
+
+        // NOTE: we're creating a libcontainer::container and calling it `sandbox`,
+        // so that's what goes in the cache.  but i think the rest of this service
+        // differentiates between the outer "pod" and the inner "container"s so we
+        // might want to do the same.
+        sandboxes.add(sandbox_id.clone(), sandbox.expect("a sandbox"))?;
```

Comment:
So I think we want a new `Sandbox` struct that contains a set of containers inside 🙁 

I think the members of the struct should be:
 
 - Init container (in this case the freshly spawned auraed)
 - A []set of tenant containers that sit along side the init container
 - Probably more in the future?